src/main/java/com/theelixrlabs/healthcare/service/PatientService.java

```diff
@@ -114,4 +133,25 @@ public PatientDTO getPatientById(UUID patientId) throws CustomException {
                 .patientAadhaarNumber(patientModel.getPatientAadhaarNumber())
                 .build();
     }
+
+    /**
+     * Deletes a patient by their ID, if conditions are met.
+     *
+     * @param patientId The ID of the patient to delete.
+     * @return A success message upon successful deletion.
+     * @throws CustomException If the patient is not found or is currently assigned to a doctor.
+     */
+    public String deletePatientById(String patientId) throws CustomException {
+        UUID validPatientId = validateUUID(patientId);
+        Optional<PatientModel> patientModelOptional = patientRepository.findById(validPatientId);
+        if (patientModelOptional.isEmpty())
+            throw new CustomException(PatientConstants.PATIENT_NOT_FOUND_KEY, messageSource);
+        List<DoctorPatientAssignmentModel> patientAssignmentList = doctorPatientAssignmentRepository.findByPatientId(validPatientId);
```

Comment:
I feel 149 to 152 u can move to separate method isPatientAssignedToDoctor() which returns boolean 
so the code can be reused for various operations

Reply:
created a separate method. Resolved.